test/e2e/gitops_service_test.go

```diff
@@ -237,3 +244,97 @@ func tearDownArgoCD(t *testing.T) {
 	assertNoError(t, err)
 
 }
+
+func policyRulesForClusterConfig() []rbacv1.PolicyRule {
```

Comment:
@amitkrout, the policy rules defined here are older than the ones specified in ArgoCD Operator OpenShift hooks. 
I don't know if it's the best way to incorporate the policy rules in e2e tests. Maybe we can export the function (https://github.com/argoproj-labs/argocd-operator/blob/master/pkg/reconciler/openshift/openshift.go#L134) in ArgoCD Operator which could be utilized directly by the e2e tests so we don't need to maintain this at both places. 

Although the best way would be to move the OpenShift specific logic in ArgoCD operator to GitOps Opeartor

Reply:
Ummm... We shouldn't be using an exported function - if we do so this test will always pass. 

If policy rules change when a new version of argocd operator controller is vendor'd in, I would expect this test to fail and then explicitly fixed.

What do you think?

Reply:
The function only returns the intended policy rules that should be specified in the Role, the actual test case is to verify the policy rules specified in the created Role with the intended policy rules, so I think it is ok to use the exported function to retrieve them. 
Also, the fix for it would be to copy the exact same rules from the ArgoCD Operator version imported as a vendor. 
So I was thinking, to avoid us the pain of updating the e2e tests every time we update the policy rules in ArgoCD operator, we can directly call the function.